packages/blocks/src/api/parser/index.js

```diff
@@ -0,0 +1,267 @@
+	const rawAttributes = rawBlock.attrs || {};
+	const rawInnerBlocks = rawBlock.innerBlocks || [];
+	let rawInnerHTML = rawBlock.innerHTML.trim();
+
+	// Fallback content may be upgraded from classic content expecting implicit
+	// automatic paragraphs, so preserve them. Assumes wpautop is idempotent,
+	// meaning there are no negative consequences to repeated autop calls.
+	if ( rawBlockName === fallbackBlockName ) {
+		rawInnerHTML = autop( rawInnerHTML ).trim();
+	}
+
+	return {
+		...rawBlock,
+		blockName: rawBlockName,
+		attrs: rawAttributes,
+		innerHTML: rawInnerHTML,
+		innerBlocks: rawInnerBlocks,
+	};
+}
+
+/**
+ * Uses the "unregistered blockType" to create a block object.
+ *
+ * @param {WPRawBlock} rawBlock block.
+ *
+ * @return {WPRawBlock} The unregistered block object.
+ */
+function createMissingBlockType( rawBlock ) {
+	const unregisteredFallbackBlock =
+		getUnregisteredTypeHandlerName() || getFreeformContentHandlerName();
+
+	// Preserve undelimited content for use by the unregistered type
+	// handler. A block node's `innerHTML` isn't enough, as that field only
+	// carries the block's own HTML and not its nested blocks.
+	const originalUndelimitedContent = serializeRawBlock( rawBlock, {
+		isCommentDelimited: false,
+	} );
+
+	// Preserve full block content for use by the unregistered type
+	// handler, block boundaries included.
+	const originalContent = serializeRawBlock( rawBlock, {
+		isCommentDelimited: true,
+	} );
+
+	return {
+		blockName: unregisteredFallbackBlock,
+		attrs: {
+			originalName: rawBlock.blockName,
+			originalContent,
+			originalUndelimitedContent,
+		},
+		innerHTML: rawBlock.blockName ? originalContent : rawBlock.innerHTML,
+		innerBlocks: rawBlock.innerBlocks,
+		innerContent: rawBlock.innerContent,
+	};
+}
+
+/**
+ * Given a raw block returned by grammar parsing, returns a fully parsed block.
+ *
+ * @param {WPRawBlock} rawBlock The raw block object.
+ *
+ * @return {WPBlock} Fully parsed block.
+ */
+export function parseRawBlock( rawBlock ) {
+	let normalizedBlock = normalizeRawBlock( rawBlock );
+
+	// During the lifecycle of the project, we renamed some old blocks
+	// And transformed others to new blocks, to avoid breaking existing content
+	// We added this function to properly parse the old content.
+	normalizedBlock = convertLegacyBlocks( normalizedBlock );
+
+	// Try finding the type for known block name.
+	let blockType = getBlockType( normalizedBlock.blockName );
+
+	// If not blockType is found for the specified name, fallback to the "unregistedBlockType".
+	if ( ! blockType ) {
+		normalizedBlock = createMissingBlockType( normalizedBlock );
+		blockType = getBlockType( normalizedBlock.blockName );
+	}
+
+	// If it's an empty freeform block or there's no blockType (no missing block handler)
+	// Then, just ignore the block.
+	// It might be a good idea to throw a warning here.
+	// TODO: I'm unsure about the unregisteredFallbackBlock check,
```

Comment:
basically the check here says: "if I'm a block with empty html and I'm a fallback block, then just remove that block, ignore it". This works for the "classic/freeform block" because it's just an empty HTML but for an "unregisteredFallbackBlock" I'm not sure, because a third-party block could register a dynamic block (which doesn't have any HTML) and then the plugin gets disabled which means the `unregisteredFallbackBlock` will be used and in this situation we still want to show that a block is missing and not just ignore it.

Reply:
Thanks for explaining!